objectstorage/objectstorage.go

```diff
@@ -0,0 +1,92 @@
+package objectstorage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"go.vocdoni.io/dvote/log"
+)
+
+type ObjectStorageClient struct {
+	config *minio.Options
+	client *minio.Client
+	bucket *minio.BucketInfo
+}
+
+// New initializes a new ObjectStorageClient with the provided API credentials and configuration.
+// It sets up a MinIO client and verifies the existence of the specified bucket.
+func New(apiKey, apiSecret, apiEndpoint, apiRegion, apiBucket string) (*ObjectStorageClient, error) {
+	config := &minio.Options{
+		Creds:  credentials.NewStaticV4(apiKey, apiSecret, ""),
+		Secure: true,
+		Region: apiRegion,
+	}
+	// Initialize minio client object.
+	minioClient, err := minio.New(apiEndpoint, config)
+	if err != nil {
+		return nil, fmt.Errorf("error seting up s3 session %v", err)
+	}
+	if exists, err := minioClient.BucketExists(context.Background(), apiBucket); err != nil || !exists {
+		return nil, fmt.Errorf("bucket %s not found: %v", apiBucket, err)
+	}
+	if policy, err := minioClient.GetBucketPolicy(context.Background(), apiBucket); err == nil {
+		log.Debugf("bucket policy: %s", policy)
+	}
+	return &ObjectStorageClient{
+		config: config,
+		client: minioClient,
+		bucket: &minio.BucketInfo{
+			Name: apiBucket,
+		},
+	}, nil
+}
+
+// Put uploads a file to the object storage service.
+func (osc *ObjectStorageClient) Put(
```

Comment:
Few suggestions:
* The comment is quite short for this method, I miss some information about the inputs and their types.
* I suggest that the object can be included into a struct to wrap the `key`, `type`, `size` and `metadata`. It can also include the content when it is returned, for example as `Get` method result.
* If `contentDisposition` has some possible values, they can be public constans or even create multiple `Put` methods: `PutInline`, ...